src/main/java/gyro/aws/codebuild/CodebuildProjectArtifacts.java

```diff
@@ -0,0 +1,215 @@
+    }
+
+    public void setEncryptionDisabled(Boolean encryptionDisabled) {
+        this.encryptionDisabled = encryptionDisabled;
+    }
+
+    /**
+     * The build output artifact location.
+     */
+    @Updatable
+    public String getLocation() {
+        return location;
+    }
+
+    public void setLocation(String location) {
+        this.location = location;
+    }
+
+    /**
+     * The pattern that is used to name and store the output artifact.
+     */
+    @Updatable
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * The pattern that is used to determine the name and location to store the output artifact.
+     */
+    @Updatable
+    @ValidStrings({ "NONE", "BUILD_ID" })
+    public String getNamespaceType() {
+        return namespaceType;
+    }
+
+    public void setNamespaceType(String namespaceType) {
+        this.namespaceType = namespaceType;
+    }
+
+    /**
+     * The field that specifies if a name specified in the buildspec file overrides the artifact name.
+     */
+    public Boolean getOverrideArtifactName() {
+        return overrideArtifactName;
+    }
+
+    public void setOverrideArtifactName(Boolean overrideArtifactName) {
+        this.overrideArtifactName = overrideArtifactName;
+    }
+
+    /**
+     * The type of build output artifact to create.
+     */
+    @Updatable
+    @ValidStrings({ "NONE", "ZIP" })
+    public String getPackaging() {
+        return packaging;
+    }
+
+    public void setPackaging(String packaging) {
+        this.packaging = packaging;
+    }
+
+    /**
+     * The pattern that is used to name and store the output artifact.
+     */
+    @Updatable
+    public String getPath() {
+        return path;
+    }
+
+    public void setPath(String path) {
+        this.path = path;
+    }
+
+    /**
+     * The identifier for this artifact definition.
+     */
+    @Output
+    public String getArtifactIdentifier() {
+        return artifactIdentifier;
+    }
+
+    public void setArtifactIdentifier(String artifactIdentifier) {
+        this.artifactIdentifier = artifactIdentifier;
+    }
+
+    @Override
+    public void copyFrom(ProjectArtifacts model) {
+        setArtifactIdentifier(model.artifactIdentifier());
+        setEncryptionDisabled(model.encryptionDisabled());
+        setLocation(model.location());
+        setName(model.name());
+        setNamespaceType(model.namespaceTypeAsString());
+        setOverrideArtifactName(model.overrideArtifactName());
+        setPackaging(model.packagingAsString());
+        setPath(model.path());
+        setType(model.typeAsString());
+
+    }
+
+    @Override
+    public String primaryKey() {
+        return "";
+    }
+
+    @Override
+    public List<ValidationError> validate(Set<String> configuredFields) {
+        List<ValidationError> errors = new ArrayList<>();
+
+        if (getLocation() == null && getType().equals("S3")) {
```

Comment:
Better that just checking null, as we also don't want blank strings in this case
```suggestion
        if (!ObjectUtils.isBlank(getLocation()) && getType().equals("S3")) {
```